IntervalTree.h

```diff
@@ -158,6 +158,18 @@ class IntervalTree {
         assert(is_valid().first);
     }
 
+    Scalar min() const {
+        assert(!empty());
+        if (left) { return left->min(); }
+        return std::min_element(intervals.begin(), intervals.end(),
+                                IntervalStartCmp())->start;
+    }
+    Scalar max() const {
```

Comment:
Is this right, or do we need to do max of `right->max()` and everything in this tree node?